src/praisonai/praisonai/bots/_protocol_mixin.py

```
@@ -254,3 +254,228 @@ def fire_message_sent(
+
+    try:
+        loop = asyncio.get_running_loop()
+    except RuntimeError:
+        loop = None
+
+    try:
+        if loop is not None and loop.is_running():
+            loop.create_task(runner.execute(event, input_data))
+        else:
+            runner.execute_sync(event, input_data)
+    except Exception as e:  # pragma: no cover — defensive
```

Ruff (0.15.18)

[warning] Do not catch blind exception: `Exception`

(BLE001)